drivers/counter/counter_ite_it51xxx.c

```diff
@@ -0,0 +1,368 @@
+	.cancel_alarm = counter_it51xxx_cancel_alarm,
+	.set_top_value = counter_it51xxx_set_top_value,
+	.get_top_value = counter_it51xxx_get_top_value,
+};
+
+#define COUNTER_IT51XXX_INIT(inst)                                                                 \
+	static void counter_it51xxx_cfg_func_##inst(const struct device *dev)                      \
+	{                                                                                          \
+		IRQ_CONNECT(DT_INST_IRQN_BY_IDX(inst, 0), 0, counter_it51xxx_alarm_isr,            \
+			    DEVICE_DT_INST_GET(inst), 0);                                          \
+		IRQ_CONNECT(DT_INST_IRQN_BY_IDX(inst, 1), 0, counter_it51xxx_top_isr,              \
+			    DEVICE_DT_INST_GET(inst), 0);                                          \
+	}                                                                                          \
+                                                                                                   \
+	static struct counter_it51xxx_config counter_it51xxx_config_##inst = {                     \
+		.info =                                                                            \
+			{                                                                          \
+				.max_top_value = UINT32_MAX,                                       \
+				.freq = 32768,                                                     \
+				.flags = 0,                                                        \
+				.channels = 1,                                                     \
+			},                                                                         \
+		.base = DT_INST_REG_ADDR(inst),                                                    \
+		.alarm_irq = DT_INST_IRQN_BY_IDX(inst, 0),                                         \
+		.alarm_flag = DT_INST_IRQ_BY_IDX(inst, 0, flags),                                  \
+		.top_irq = DT_INST_IRQN_BY_IDX(inst, 1),                                           \
+		.top_flag = DT_INST_IRQ_BY_IDX(inst, 1, flags),                                    \
+		.irq_config_func = counter_it51xxx_cfg_func_##inst,                                \
+	};                                                                                         \
+                                                                                                   \
+	static struct counter_it51xxx_data counter_it51xxx_data_##inst;                            \
+                                                                                                   \
+	DEVICE_DT_INST_DEFINE(inst, &counter_it51xxx_init, NULL, &counter_it51xxx_data_##inst,     \
+			      &counter_it51xxx_config_##inst, POST_KERNEL,                         \
+			      CONFIG_COUNTER_INIT_PRIORITY, &counter_it51xxx_driver_api);
+
+DT_INST_FOREACH_STATUS_OKAY(COUNTER_IT51XXX_INIT)
+BUILD_ASSERT(DT_NUM_INST_STATUS_OKAY(DT_DRV_COMPAT) == 1,
+	     "only one ite,it51xxx-counter compatible node can be supported");
```

Comment:
is that true? looks like everything is in place to support multiple instances

Reply:
Yes, so far, timer2, timer7, and timer8 of it51xxx can be used. After implementing counter driver, there is a left single timer2. Due to our hardware design, we need two timers for each counter.